bn-apps/memberships-management/membership-management-demo/src/main/kotlin/com/r3/businessnetworks/memberships/demo/flows/Flows.kt

```diff
@@ -0,0 +1,124 @@
+package com.r3.businessnetworks.memberships.demo.flows
+
+import co.paralleluniverse.fibers.Suspendable
+import com.r3.businessnetworks.membership.flows.bno.ActivateMembershipFlow
+import com.r3.businessnetworks.membership.flows.bno.service.DatabaseService
+import com.r3.businessnetworks.membership.flows.member.GetMembershipsFlow
+import com.r3.businessnetworks.membership.flows.member.MembershipListRequest
+import com.r3.businessnetworks.membership.flows.member.MembershipsListResponse
+import com.r3.businessnetworks.membership.flows.member.RequestMembershipFlow
+import com.r3.businessnetworks.membership.flows.member.service.MembershipsCacheHolder
+import com.r3.businessnetworks.membership.states.MembershipContract
+import com.r3.businessnetworks.membership.states.MembershipState
+
+
+import com.r3.businessnetworks.memberships.demo.contracts.SampleContract
+import com.r3.businessnetworks.memberships.demo.contracts.SampleState
+import net.corda.core.contracts.ReferencedStateAndRef
+import net.corda.core.contracts.StateAndRef
+import net.corda.core.contracts.requireThat
+import net.corda.core.flows.*
+import net.corda.core.identity.Party
+import net.corda.core.serialization.CordaSerializable
+import net.corda.core.transactions.SignedTransaction
+import net.corda.core.transactions.TransactionBuilder
+import net.corda.core.utilities.unwrap
+
+@CordaSerializable
+class RequestMembershipForTransaction (val bno : Party)
+
+class IssueAssetFlow (val bno : Party) : FlowLogic<SignedTransaction>() {
+    @Suspendable
+    override fun call() : SignedTransaction {
+        // for the sake of demo we are just taking the first available notary
+        val notary = serviceHub.networkMapCache.notaryIdentities.single()
+
+        // query the vault with bno
+        val memberships= subFlow(GetMembershipsFlow(bno))
+        val ourMembership = memberships[ourIdentity]!!
```

Comment:
This will throw `NullPointerException` if membership is missing, which isn't very user friendly. In kotlin you can use the following syntax `val ourMembership = membership[ourIdentity] ?: throw FlowException("Membership for $ourIdentitiy has not been found")`

Reply:
nice. done